scipy/stats/_distribution_infrastructure.py

```diff
@@ -395,13 +406,22 @@ class _RealDomain(_SimpleDomain):
     """
 
     def __str__(self):
-        a, b = self.endpoints
-        left_inclusive, right_inclusive = self.inclusive
+        if callable(self.endpoints):
+            if self.endpoints.__doc__ is not None:
+                return self.endpoints.__doc__
+            # If endpoints callable has no docstring, fall back to a generic
+            # str which only expresses that the endpoints equal some function
+            # of the parameters.
+            params = inspect.getfullargspec(self.endpoints).kwonlyargs
+            a, b = f"f({','.join(params)})", f"g({','.join(params)})"
```

Comment:
Is it overkill to bring in `inspect` just to have a better default `__str__` for domains? I'm happy to get rid of this.